core/src/main/scala/org/trustedanalytics/sparktk/frame/internal/ops/flatten/FlattenColumns.scala

```diff
@@ -0,0 +1,69 @@
+package org.trustedanalytics.sparktk.frame.internal.ops.flatten
+
+import org.apache.commons.lang3.StringUtils
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Row
+import org.trustedanalytics.sparktk.frame.DataTypes
+import org.trustedanalytics.sparktk.frame.internal.{ FrameState, FrameTransform, BaseFrame }
+
+trait FlattenColumnsTransform extends BaseFrame {
+
+  def flattenColumns(columns: List[String],
+                     delimiters: Option[List[String]]): Unit = {
+    execute(FlattenColumns(columns, delimiters))
+  }
+}
+
+/**
+ * Spread data to multiple rows based on cell data.
+ *
+ * @param columns The columns to be flattened.
+ * @param delimiters The list of delimiter strings for each column.  Default is comma (,).
+ */
+case class FlattenColumns(columns: List[String],
+                          delimiters: Option[List[String]]) extends FrameTransform {
+  require(columns != null && columns.nonEmpty, "column list is required")
+  columns.foreach(x => require(StringUtils.isNotBlank(x), "non empty column names are required."))
+
+  // If just one delimiter is defined, use the same delimiter for all columns.  Otherwise, use the default.
+  lazy val defaultDelimiters = Array.fill(columns.size) { if (delimiters.isDefined && delimiters.get.size == 1) delimiters.get(0) else "," }
+
+  override def work(state: FrameState): FrameState = {
+    var flattener: RDD[Row] => RDD[Row] = null
+    val columnIndices = columns.map(c => state.schema.columnIndex(c))
+    val columnDataTypes = columns.map(c => state.schema.columnDataType(c))
```

Comment:
I learned this recently...

{code}
val (columnIndices, columnDataTypes) = columns.map(c => (state.schema.columnIndex(c), state.schema.columnDataType(c))).unzip
{code}
but I'm not sure it makes the code any more readable or any better performant